app/Helpers/database/game.php

```diff
@@ -998,6 +1012,29 @@ function GetPatchData(int $gameID, ?User $user, int $flag): array
         'Leaderboards' => [],
     ];
 
+    if ($clientSupportLevel !== ClientSupportLevel::Full) {
+        if ($game->achievements_published < 0) { // will never be true. change to > when ready
```

Comment:
If you want to test this functionality, you can change the `<` to a `>` and point an older client at the test site.

The player should "unlock" the warning achievement a few seconds after loading the game:
![image](https://github.com/user-attachments/assets/8141dbdc-c38c-4714-b91d-4a2f763bff5b)

The achievement will also appear in their achievement list as a reminder.
![image](https://github.com/user-attachments/assets/9eaac8a3-d9bd-4169-8ccd-e2326989bec3)

The achievement will automatically be marked as unlocked for softcore, so the warning should only appear when the player has hardcore enabled. The achievement will still appear in the list when playing in softcore.